require fully qualified image names

dustinkirkland opened the pull request:
**Dimitri John Ledkov**:  all tooling in containerd, crane, docker, chainctl expects full names. And simple name is always expected to be resolved against locally created or tagged image. Please do not do heuristics which are not what all other containers tools do when specifying image names

xnox reviewed (changes requested):
not sure it is working

```
$ ./chainguard-source --image cgr.dev/chainguard-private/python

This tool will consume a lot of network and disk, as it fetches full
git source code repositories and source archive files from remote
servers (outside of chainguard.dev).

You will likely need several GB of free disk space in your current
working directory [/home/xnox/chainguard-dev/chainguard-source]:
Filesystem                         Size  Used Avail Use% Mounted on
/dev/mapper/ubuntu--vg-ubuntu--lv  618G  476G  111G  82% /

And this will consume a lot of network bandwith.
Your connection may be throttled by the remote servers.

Do you want to continue? [y/n]: y
INFO: Fetching sbom [amd64/cgr.dev/chainguard-private/python]
./chainguard-source: line 335: /home/xnox/chainguard-dev/chainguard-source/sources/cgr.dev/chainguard-private/python-amd64/sboms/cgr.dev/chainguard-private/python-amd64.sbom.spdx.json: No such file or directory

```

Maybe path should be escaped or encoded or created somehow?

```
$ systemd-escape cgr.dev/chainguard-private/python
cgr.dev-chainguard\x2dprivate-python
```

could do a trick, but other options are available too, for example urlencode?

also existing public examples don't work either as per updated readme.

dustinkirkland commented:
Okay, thanks, Dimitri.  I fixed that and pushed the commits.  I removed the architecture from the directory names, which was causing this problem.  I'm running this now, and it's currently working...  All the way up to the point that we try to clone a private repository over git https.  At which point that fails, as its needing a github credential.  So I'll need to reset and mangle the git urls and clone over SSH rather than HTTPS....

dustinkirkland commented:
Okay, ready for a new review and testing, @xnox !  This is working for me now....  Takes about 5 minutes.  There are some download optimizations, if we want to look at later.  But this is now working, end to end.

```
$ ./chainguard-source -y --image cgr.dev/chainguard-private/python-fips
...
$ du -sh sources
5.5G  .
```
